dsgrid/config/project_config.py

```diff
@@ -166,6 +232,11 @@ class InputDatasetModel(DSGBaseModel):
         updateable=False,
         # TODO: add notes about warnings for outdated versions? DSGRID-189.
     )
+    required_dimensions: RequiredDimensionsModel = Field(
+        title="required_dimenions",
+        description="Defines required record IDs that must exist for each dimension.",
```

Comment:
No, any combination of base or supplemental dimension _record_ ID.

Reply:
Here is a specific example:

In this paradigm StandardScenarios would define this supplemental dimension:
```
name = "residential-subsectors"
display_name = "Residential Subsectors"
dimension_query_name = "residential_subsectors"
type = "subsector"
id = "residential-subsectors__656c0d22-c47e-4f95-a96e-1e31c17d8d16"
```
It would have one record:
```
id,name
residential_subsectors,Residential Subsectors
```
It would have these mapping records:
```
from_id,to_id
mobile_home,residential_subsectors
multifamily_2_to_4_units,residential_subsectors
multifamily_5_plus_units,residential_subsectors
single_family_attached,residential_subsectors
single_family_detached,residential_subsectors
```
`name`, `id`, `dimension_query_name`, `display_name`, and `record_id` are all very similar.

Reply:
Oh--that is weirder than I thought. I was thinking we could list the supplemental name, id, or query_name so that any base dimension record id that gets mapped to that supplemental dimension would be included.

Being able to reference any supplemental dimension record id also seems to set us up for name conflicts/ambiguities.

Per comments above, maybe we should use different data models for specifying requirements via (a) base dimension record ids versus (b) supplemental dimensions. The latter needs to identify the specific dimension (by name, id, or query name) and then could potentially also let users specify specific record ids.